raft/raft_test.go

```diff
@@ -356,8 +356,8 @@ func TestLearnerPromotion(t *testing.T) {
 
 	nt.send(pb.Message{From: 1, To: 1, Type: pb.MsgBeat})
 
-	n1.addNode(2)
```

Comment:
nit: for the purpose of avoiding verbosity and the diff, you might consider adding `addNode`, `removeNode`, and `addLearner` as test-only (in `raft_test.go`) methods. They would be one-liners.

Reply:
Not a bad idea, but I'll keep as is.